components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -331,7 +344,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             );
 
         for (const projectId of projects) {
-            this.disposables.pushAll([this.subscriber.listenForPrebuildUpdates(projectId, handler)]);
+            this.disposables.push(this.subscriber.listenForPrebuildUpdates(projectId, handler));
```

Comment:
here disposables can be already disposed because of async call above?

Reply:
Nice catch! I think a `DisposableCollection` is supposed to be prefilled with single item first, so that if it's emptied once, that state should be kept consistent.

Reply:
Thanks @svenefftinge and @akosyakov for checking on this. 

The PR is updated now, with a different set of changes:
* check for `client` be defined in gitpod-server-impl
* add prebuild subscribers only if the thing is not disposed yet
   * this might have happened frequently on very short living `workspacePageClose` events